pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go

```diff
@@ -19,35 +18,43 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/streaming"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/errors"
+	"github.com/jackc/pgx/v4"
 )
 
 type partitionedStreamClient struct {
-	srcDB          *gosql.DB // DB handle to the source cluster
 	urlPlaceholder url.URL
+	pgxConfig      *pgx.ConnConfig
 
 	mu struct {
 		syncutil.Mutex
 
 		closed              bool
 		activeSubscriptions map[*partitionedStreamSubscription]struct{}
+		srcConn             *pgx.Conn // pgx connection to the source cluster
 	}
 }
 
 func newPartitionedStreamClient(remote *url.URL) (*partitionedStreamClient, error) {
-	db, err := gosql.Open("postgres", remote.String())
+	config, err := pgx.ParseConfig(remote.String())
+	if err != nil {
+		return nil, err
+	}
+	conn, err := pgx.ConnectConfig(context.Background(), config)
```

Comment:
Do we want to use background ctx here? Seems like piping ctx is not much of a lift; 
I don't know what pgx does in the face of context cancellation; but it definitely does nothing w/ background context.